pallets/parachain-staking/src/lib.rs

```diff
@@ -2014,6 +2052,25 @@ pub mod pallet {
 
 			Ok(Some(<T as pallet::Config>::WeightInfo::unlock_unstaked(unstaking_len)).into())
 		}
+
+		// TODO: Benchmark, unit tests, docs
+		#[pallet::weight(0)]
+		pub fn claim_rewards_for(origin: OriginFor<T>, target: <T::Lookup as StaticLookup>::Source) -> DispatchResult {
```

Comment:
Any specific reason why we would allow anyone to trigger a payout for anyone else? I still don't understand what would be the use case for this, if I want to have control over when I would claim my rewards. If I want to delegate, I could set up a proxy,

Reply:
IMO, the main reason would be to cleanup storage and consume tiny rewards that have not been claimed by potentially exited users. I understand that's a very unlikely scenario and each user has an incentive to claim their rewards.

Moreover, in frame staking this is also possible by triggering a payout for a validator which rewards their nominators as well.

Reply:
I am completely fine changing this and would be interested in @weichweich 's opinion.

Reply:
Yeah I think Parity does that because otherwise users would just forget and keep piling up stuff, I think. Which would be fine in our case if that is the assumption. But of course the downside is that I can't simply decide to trigger my reward at any time, because someone else inside my delegation pool might do it for me.

Reply:
Let's first start with the more restrictive version. We might be able to change that later if we feel the need for it.

Reply:
By more restrictive you are referring to Antonio's proposal, e.g. origins can only claim for themselves?

Reply:
[420170d](https://github.com/KILTprotocol/kilt-node/pull/384/commits/420170d387972e64c0e0c50997f8ca1110cac9f4)